catkin_ws/src/kpr_arm_control/src/arm_control/arm_control.cpp

```diff
@@ -90,33 +110,140 @@ bool sendIO(int8_t pin, float state) {
 }
 
 /**
+ * Function which checks whether the previously changed output has the desired effect.
+ * It makes sure that the value checked is indeed the response of the new sensor and 
+ * not the old response, since it is internally updated at f = 10 Hz. 
+ * 
+ * @param sensor The sensor which output needs to be read.
+ * @param pinstates[] Vector of the outputs, which contains the latest data from the sensors.
+ * 
+ * @return Passes the boolean value received from the sensors.
+ */
+bool readDigInput(int sensor, bool pinstates[]) {
```

Comment:
it is not needed to give the pinstates variable as an argument of this function, as it is already in scope